### PR TITLE
feat: split and export renovate presets

### LINKED ISSUE #1483: [Feature]: Provide Renovate Presets for Otter recommended rules
### Context

Today Renovate is used within the Otter Project package and in the project generated by the Otter generators.
The Renovate config used and generated is based on Renovate out-of-the-box preset and add, on top of it, rules dedicated to Otter packages and recommended by the Otter guideline.

### Proposal

The feature should contain the following items:
- [ ] Expose Renovate Preset for Otter packages (grouping, default upgrade strategy, etc...)
- [ ] Expose Renovate Presets for different grouping (such as  Angular, Design Factory, non-CJS exposing packages, etc...)
- [ ] Expose Renovate Preset for SDK generation
- [ ] Provide `ngUpdate` to add the presets to existing Otter projects (without touching rule override)

> [!NOTE]
> The presets should be exposed at the root level of the Otter Github repository to be easy used by the projects.

### PR DESCRIPTION
**Pull request by kpanot**
## Proposed change

Split and export renovate presets

## Related issues

- :rocket: Feature resolves #1483

### CODE REVIEW

File: docs/renovate/readme.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Renovate Presets
+
+The Otter library provide a set of [Renovate Presets](https://docs.renovatebot.com/config-presets/) custom to facilitate the configuration of automated dependency updates.
+
+## Usage
+
+To be able to benefice of the Renovate Presets exposed by the Otter Framework, you can add the configurations in the `extends` field in your `.renovaterc.json` file as described in the following example:
```
Comment (fpaul-1A) [inline docs/renovate/readme.md:7]:
```suggestion
To be able to benefit from the Renovate Presets exposed by the Otter Framework, you can add the configurations in the `extends` field in your `.renovaterc.json` file as described in the following example:
```

File: docs/renovate/readme.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Renovate Presets
+
+The Otter library provide a set of [Renovate Presets](https://docs.renovatebot.com/config-presets/) custom to facilitate the configuration of automated dependency updates.
```
Comment (fpaul-1A) [inline docs/renovate/readme.md:3]:
```suggestion
The Otter library provides a set of custom [Renovate Presets](https://docs.renovatebot.com/config-presets/) to facilitate the configuration of automated dependency updates.
```

File: docs/renovate/readme.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+}
+```
+
+> [!NOTE]
+> The Otter Renovate presets are always prefixed with `github>AmadeusITGroup/otter//tools/renovate/` to target the specific configuration files hosted in the Otter repository.
+
+## Available presets
+
+- **base**: Base configuration recommended to an Otter base project.
+- **otter-project**: Configuration to used in a project to ensure, facilitate and regroup the upgrades of the Otter dependencies.
```
Comment (fpaul-1A) [inline docs/renovate/readme.md:28]:
```suggestion
- **otter-project**: Configuration to be used in a project to facilitate and regroup the upgrades of Otter dependencies.
```

File: tools/renovate/otter-project.json [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        "@ama-sdk/core",
+        "@otter/core"
+      ],
+      "postUpgradeTasks": {
+        "commands": [
+          "npm install",
+          "npm run ng update {{{depName}}} --from={{{currentVersion}}} --to={{{newVersion}}} --migrate-only --allow-dirty --force"
+        ],
+        "fileFilters": [
+          "**"
```
Comment (fpaul-1A) [inline tools/renovate/otter-project.json:55]: this won't include files/folders starting with a `.`

Comment (kpanot) [reply tools/renovate/otter-project.json]: just copy pasted the rule from the original config but I fixed it

**Review by fpaul-1A — APPROVED** [review]

File: packages/@o3r/workspace/schematics/ng-add/helpers/renovate/templates/__dot__renovaterc.json.template [hunk 5]
```diff
@@ -2,15 +2,13 @@
...
-    "group:recommended",
-    "group:test"
+    "github>AmadeusITGroup/otter//tools/renovate/base",
+    "github>AmadeusITGroup/otter//tools/renovate/otter-project"
   ],
   "prConcurrentLimit": 0,
   "prHourlyLimit": 0,
   "baseBranches": [
-    "master"
+    "main"
```
Comment (cpaulve-1A) [inline packages/@o3r/workspace/schematics/ng-add/helpers/renovate/templates/__dot__renovaterc.json.template:11]: why change from master to main like this?

File: tools/renovate/otter-project.json [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        "@ama-sdk/core",
+        "@otter/core"
+      ],
+      "postUpgradeTasks": {
+        "commands": [
+          "yarn install",
+          "yarn run ng update {{{depName}}} --from={{{currentVersion}}} --to={{{newVersion}}} --migrate-only --allow-dirty --force"
+        ],
+        "fileFilters": [
+          "!**/.npmrc"
```
Comment (cpaulve-1A) [inline tools/renovate/otter-project.json:35]: If we need to patch the yarnrc to run the install and subsequent script, the yarnrc should also be excluded from the files to commit
```suggestion
          "!**/.{npmrc,yarnrc*}"
```

File: tools/renovate/sdk-spec-upgrade.json [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+        "main"
+      ],
+      "postUpgradeTasks": {
+        "commands": [
+          "npm install",
+          "npm run swagger:generate"
+        ],
+        "executionMode": "branch",
+        "fileFilters": [
+          "!**/.npmrc"
```
Comment (cpaulve-1A) [inline tools/renovate/sdk-spec-upgrade.json:38]: Same remark to add the yarnrc file

File: tools/renovate/sdk.json [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+      "matchPackageNames": [
+        "npm"
+      ],
+      "postUpgradeTasks": {
+        "commands": [
+          "npm install",
+          "npm run swagger:generate"
+        ],
+        "fileFilters": [
+          "!**/.npmrc"
```
Comment (cpaulve-1A) [inline tools/renovate/sdk.json:62]: Same remark for yarnrc

File: tools/renovate/sdk.json [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+      "matchPackageNames": [
+        "yarn"
+      ],
+      "postUpgradeTasks": {
+        "commands": [
+          "yarn install",
+          "yarn run swagger:generate"
+        ],
+        "fileFilters": [
+          "!**/.npmrc"
```
Comment (cpaulve-1A) [inline tools/renovate/sdk.json:85]: Same remark for yarnrc

File: packages/@o3r/workspace/schematics/ng-add/helpers/renovate/templates/__dot__renovaterc.json.template (hunk 5 above)
Comment (kpanot) [reply packages/@o3r/workspace/schematics/ng-add/helpers/renovate/templates/__dot__renovaterc.json.template:11]: To be align with `github` and `bitbucket` default base branch name.

**Review by cpaulve-1A — APPROVED** [review]

**Review by fpaul-1A — APPROVED** [review]

**Pull request merged**